File: ArmstrongNumbers/ArmstrongNumbers.php
<?php

function IfArmstrongNumber(int $possibleArmstrongNumber = 100): bool
{
    $sumOfCubeOfDigits = array_sum(
        array_map(
            function ($digit) use($possibleArmstrongNumber){
                return pow($digit,strlen((string)$possibleArmstrongNumber));
            },
            str_split((string)$possibleArmstrongNumber)
        )
    );

    return ($sumOfCubeOfDigits === $possibleArmstrongNumber) ? true : false;
}

$start = 100;
$end = 1000;
echo "Armstrong numbers from {$start} to {$end}\n";

for ($number = $start; $number <= $end; $number++) {
    if (IfArmstrongNumber($number)) {
        echo $number . "\n";
    }
}

File: ArmstrongNumbers/ArmstrongNumbers.cpp
#include <iostream>
using namespace std;

int main()
{
    int start = 100, end = 1000;
    
    cout<<"Armstrong Numbers:\nfrom "<<start<<" to "<<end<<"\n";
    
    for(int AnArmstrongNumber = start; AnArmstrongNumber <= end; AnArmstrongNumber++)
    {
        int sumOfCubeOfDigit=0,digit=0,number=AnArmstrongNumber;
        
        while(number>0)
        {    
            digit = number%10;
            sumOfCubeOfDigit += (digit*digit*digit);
            number /= 10;
        }
        
        if(AnArmstrongNumber==sumOfCubeOfDigit)
        {
            cout<<AnArmstrongNumber<<"\n";
        }
    }

    return 0;
}